### ISSUE TITLE
HelpNeeded: Difference between KMeans 2023.0.0 & 2023.1.1

### ISSUE THREAD

**Issue by shashank-fujitsu** [intel/scikit-learn-intelex#1258]
**I got speedup of 10x**

For KMeans Algorithm - fit method
- In 2023.0.0 it took around 4 seconds 
- In 2023.1.1 it took around 0.5 seconds

But now, I am not able to run my script in v2023.0.0

**In Commit History**
I can see - Explicit passing of gamma parameter for kmeans (commit-id: 1dca20c3761197d082d548f27f698d6389e60fbe)

**What does GAMMA do?**
Is this the one which makes it run faster?

**OS:**
 - Ubuntu 20.04

**Comment by napetrov** [comment]
@shashank-fujitsu  
Can you please provide additional details on code that you was using and also some dataset data? 

This repo is handling scikit-learn-intelex part that focus on python API/patching. While the oneDAL is C++ lib underneath that actually do compute - https://github.com/oneapi-src/oneDAL


Also on this one - "But now, I am not able to run my script in v2023.0.0" - can you please elaborate more?

**Comment by shashank-fujitsu** [comment]
Hi @napetrov, 

Thanks for the quick reply..

I was trying to run the following script on this [Dataset](https://www.kaggle.com/datasets/vijayuv/onlineretail).

```
import numpy as np
import pandas as pd
import time
from sklearnex import patch_sklearn
patch_sklearn()
from sklearn.cluster import KMeans

df = pd.read_csvl('Online Retail.csv')
print("finished reading")

ft = np.array(df.UnitPrice)
print(ft.size)

start = time.time()
kmeans = KMeans(n_clusters=200, random_state=0).fit(rft)
end = time.time()
print(f"Training Time taken: {(end-start)*10**3:.03f}ms")
```
It's much faster in 2023.1.1 as compared to 2023.0.0

When I am trying to run the same script now in 2023.0.0
I get the following error,

```
Traceback (most recent call last):
  File "scikit-demo.py", line 31, in <module>
    kmeans = KMeans(n_clusters=200, random_state=0).fit(rft)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 88, in wrapper_with_self
    return wrapper_impl(self, *args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 74, in wrapper_impl
    result = _run_on_device(func, q, obj, *hostargs, **hostkwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 65, in _run_on_device
    return dispatch_by_obj(obj, func, *args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 53, in dispatch_by_obj
    return func(obj, *args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/daal4py/sklearn/cluster/_k_means_0_23.py", line 559, in fit
    return _fit(self, X, y=y, sample_weight=sample_weight)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/daal4py/sklearn/cluster/_k_means_0_23.py", line 375, in _fit
    _daal4py_k_means_fit(
  File "/home/ubuntu/.local/lib/python3.8/site-packages/daal4py/sklearn/cluster/_k_means_0_23.py", line 204, in _daal4py_k_means_fit
    res = kmeans_algo.compute(X, starting_centroids_)
  File "build/daal4py_cy.pyx", line 13459, in daal4py._daal4py.kmeans.compute
  File "build/daal4py_cy.pyx", line 246, in daal4py._daal4py._execute_with_context.exec_func
  File "build/daal4py_cy.pyx", line 13441, in daal4py._daal4py.kmeans._compute
RuntimeError: Incorrect parameter
Details:
Parameter name: gamma
```
Thanks, in Advance.

**Comment by shashank-fujitsu** [comment]
@napetrov Thanks for the response, this solves one of my problems.

I started getting this error now, (I am using conda env now)
```
Traceback (most recent call last):
  File "script_2.py", line 6, in <module>
    from sklearnex import patch_sklearn
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/sklearnex/__init__.py", line 18, in <module>
    from .dispatcher import patch_sklearn
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/sklearnex/dispatcher.py", line 22, in <module>
    from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/__init__.py", line 38, in <module>
    from daal4py._daal4py import *
ImportError: libsvml.so: cannot open shared object file: No such file or directory
```
Could you let me know, what I am doing wrong?


I will look forward for a response from @KulikovNikita & @Alexsandruss.

Thanks in Advance

**Comment by shashank-fujitsu** [comment]
Hi @Alexsandruss, Thanks, it works now.

Also, could you please let me know the version of stock scikit-learn I should be using for the script to run on 2023.0.0? 

Thanks

**Comment by shashank-fujitsu** [comment]
> There is no recommendation to use specific sklearn version, you can use latest available (1.2.2).

But I get this error while running the script.
```
Traceback (most recent call last):
  File "script_2.py", line 27, in <module>
    KMeans(n_clusters=200, random_state=0).fit(rft)
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 88, in wrapper_with_self
    return wrapper_impl(self, *args, **kwargs)
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 74, in wrapper_impl
    result = _run_on_device(func, q, obj, *hostargs, **hostkwargs)
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 65, in _run_on_device
    return dispatch_by_obj(obj, func, *args, **kwargs)
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/sklearn/_device_offload.py", line 53, in dispatch_by_obj
    return func(obj, *args, **kwargs)
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/sklearn/cluster/_k_means_0_23.py", line 559, in fit
    return _fit(self, X, y=y, sample_weight=sample_weight)
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/sklearn/cluster/_k_means_0_23.py", line 375, in _fit
    _daal4py_k_means_fit(
  File "/home/ubuntu/code/envs/sklearnexold/lib/python3.8/site-packages/daal4py/sklearn/cluster/_k_means_0_23.py", line 204, in _daal4py_k_means_fit
    res = kmeans_algo.compute(X, starting_centroids_)
  File "build/daal4py_cy.pyx", line 13459, in daal4py._daal4py.kmeans.compute
  File "build/daal4py_cy.pyx", line 246, in daal4py._daal4py._execute_with_context.exec_func
  File "build/daal4py_cy.pyx", line 13441, in daal4py._daal4py.kmeans._compute
RuntimeError: Incorrect parameter
Details:
Parameter name: gamma
```
What should I do?

**Comment by shashank-fujitsu** [comment]
I also tried downgrading stock scikit-learn to 0.22, I was getting the same error again.
```
RuntimeError: Incorrect parameter
Details:
Parameter name: gamma
```

**Comment by Alexsandruss** [comment]
It's bug in sklearnex/oneDAL, it was solved here - https://github.com/intel/scikit-learn-intelex/commit/1dca20c3761197d082d548f27f698d6389e60fbe, and included starting 2023.0.1 release.

**Comment by shashank-fujitsu** [comment]
@Alexsandruss , I'll go through this.

Thanks for your time, really appreciate it.
